src/mscorlib/src/System/ReadOnlySpan.cs

```diff
@@ -144,5 +164,47 @@ public bool Equals(ReadOnlySpan<T> other)
             return (_length == other._length) &&
                 (_length == 0 || JitHelpers.ByRefEquals(ref JitHelpers.GetByRef<T>(ref _rawPointer), ref JitHelpers.GetByRef<T>(ref other._rawPointer)));
         }
+
+        /// <summary>
+        /// Copies the contents of this span into destination span. The destination
+        /// must be at least as big as the source, and may be bigger.
+        /// </summary>
+        /// <param name="destination">The span to copy items into.</param>
+        public bool TryCopyTo(Span<T> destination)
+        {
+            if (Length > destination.Length)
+                return false;
+
+            SpanHelper.CopyTo<T>(ref destination._rawPointer, ref _rawPointer, Length);
+            return true;
+        }
+
+        /// <summary>
+        /// Copies the contents of this span into destination array. The destination
+        /// must be at least as big as the source, and may be bigger.
+        /// </summary>
+        /// <param name="destination">The array to copy items into.</param>
+        public bool TryCopyTo(T[] destination)
+        {
+            return TryCopyTo(new Span<T>(destination));
+        }
+
+        /// <summary>
+        /// Copies the contents of this span into destination memory. 
+        /// Only Value Types that contain no pointers are supported.
+        /// </summary>
+        /// <param name="destination">An unmanaged pointer to memory.</param>
+        /// <param name="elementsCount">The number of T elements the memory contains.</param>
+        [CLSCompliant(false)]
+        public unsafe bool TryCopyTo(void* destination, int elementsCount)
```

Comment:
I think it is confusing either way. I think it may be better to not have this unsafe API at all - the fewer APIs that take unmanaged pointers the better, and instead make people to use the other overload that takes Span - it more clearly expresses the intent:

`TryCopyTo(Span<int>(pBuffer, nElements))`.